### PR TITLE
FV: speed things up a bit

### PR DESCRIPTION
**Pull request by rwcarlsen**
This improves FV simple diffusion speed by ~4x.  I just added an FV-specific assembly reinit function and quadrature rule.  There is room for another 1.5-2x speedup going forward, but this should be a welcome improvement for the time being.  Notably, FV diffusion is still a bit slower than FE diffusion because FV needs both the face and the elemental loops - and we don't currently/yet skip the FE specific stuff in the elemental loop.  This just skips FE specific stuff in the face loop.

### CODE REVIEW

**Comment by rwcarlsen** [comment]
I've asked @snschune to make sure this doesn't cause any problems with his pronghorn work - we should wait for that confirmation before merging.

**Review by snschune — DISMISSED** [review]
Doesn't break any of my tests. 👍

File: framework/src/base/Assembly.C
Comment (lindsayad) [inline framework/src/base/Assembly.C:518]: What’s with this hard-codes size here but then you use the mesh dimension immediately below?

Comment (rwcarlsen) [reply framework/src/base/Assembly.C]: fixed

**Comment by lindsayad** [comment]
Thanks. Do you want to squash the fixup commit?

**Comment by rwcarlsen** [comment]
I squashed it (and fixed the crashes caused by my off-by-one resizing)

**Review by lindsayad — APPROVED** [review]

**Pull request merged**
